spec/unit/puppet/type/deferred_resources_spec.rb

```diff
@@ -183,6 +297,52 @@
 
         expect(@catalog.resource('Package', 'mypackage')).to be_nil
       end
+
+      it 'should not override any existing resources' do
+          @resource = deferred_resources_type.new(
+            :name                         => 'foo',
+            :resource_type                => 'file',
+            :mode                         => 'warning',
+            :override_existing_attributes => {
+              'owner'   => nil,
+              'group'   => nil,
+              'content' => {
+                'invalidates' => ['source']
+              }
+            },
+            :resources                    => {
+              '/tmp/test' => {
+                'ensure'  => 'file',
+                'owner'   => 'bob',
+                'group'   => 'alice',
+                'content' => 'Some stuff',
+                # This should be ignored
+                'mode'    => '0777'
+              }
+            }
+          )
+
+          @catalog.create_resource('File', {
+            'name'    => '/tmp/test',
+            'ensure'  => 'file',
+            'owner'   => 'root',
+            'group'   => 'root',
+            'source' => 'puppet:///my.server/test',
+            'mode'    => '0644'
+          })
+
+          Puppet.expects(:send).with(:warning, %{deferred_resources: Would have overridden attributes 'owner', 'group', 'content' on existing resource File[/tmp/test]}).once
+
+          @resource.autorequire
+
+          result = @catalog.resource('File[/tmp/test]')
+
+          expect(result[:owner]).to eq('root')
+          expect(result[:group]).to eq('root')
```

Comment:
Why weren't these overridden to be `nil`?

Reply:
`nil` means `undef` as in "ignore this".

Reply:
roger